includes/class-wc-payments-order-service.php

```diff
@@ -901,12 +901,13 @@ public function get_fraud_meta_box_type_for_order( $order ): string {
 	 *
 	 * @param WC_Order                                                          $order The order.
 	 * @param WC_Payments_API_Payment_Intention|WC_Payments_API_Setup_Intention $intent The payment or setup intention object.
+	 * @param bool                                                              $is_changing_payment_method_for_subscription Whether the payment is being changed for a subscription.
 	 *
 	 * @throws Order_Not_Found_Exception
 	 */
-	public function attach_intent_info_to_order( WC_Order $order, $intent ) {
+	public function attach_intent_info_to_order( WC_Order $order, $intent, $is_changing_payment_method_for_subscription = false ) {
 		// We don't want to allow metadata for a successful payment to be disrupted.
-		if ( Intent_Status::SUCCEEDED === $this->get_intention_status_for_order( $order ) ) {
+		if ( Intent_Status::SUCCEEDED === $this->get_intention_status_for_order( $order ) && ! $is_changing_payment_method_for_subscription ) {
```

Comment:
We should let the change payment method flow update the intent in order even though the previous intent is successful, that's why we are not letting this function to stop the flow for successful intents with change payment method use-case.